openshift/e2e-common.sh

```diff
@@ -459,6 +459,42 @@ function run_e2e_tests(){
     --skip-cleanup-on-fail \
     --resolvabledomain || failed=3
 
+  # Test gRPC via OpenShift Route.
+  # * OCP Route does not work with websocket when enabling default-enable-http2. It will be fixed in the next haproxy version (OCP 4.12 or 4.13).
+  # * Also, Skip 4.9, 4.8 job as OCP option for gRPC/HTTP2 is available since 4.10 - bz#1826225
+  if [[ ${JOB_NAME} =~ "48" ]] || [[ ${JOB_NAME} =~ "49" ]]; then
+        echo "skip gRPC test via OCP"
+        return $failed
+  fi
+
+  echo "gRPC test via OCP"
+
+  oc annotate ingresses.config/cluster ingress.operator.openshift.io/default-enable-http2=true
+  oc annotate knativeserving knative-serving -n knative-serving serverless.openshift.io/default-enable-http2=true
+
+  # This is not necessary actually but it makes usre that access passes through OCP route.
+  oc patch knativeserving knative-serving \
+      -n "${SERVING_NAMESPACE}" \
+      --type merge --patch '{"spec": {"ingress": {"kourier": {"service-type": "ClusterIP"}}}}'
+
+  if [[ $(oc get infrastructure cluster -ojsonpath='{.status.platform}') = VSphere ]]; then
```

Comment:
Why VSphere is different could you elaborate a bit?

Reply:
Because of the line above https://github.com/openshift-knative/serving/blob/806c070acba9dc899eac437d2c98136ad63348c0/openshift/e2e-common.sh#L346-L350
It is not introduced by this commit by the following commit for a long time ago - https://github.com/openshift/knative-serving/pull/490

I am not a person who made it but I guess VSphere is not cluster but sort of bare metal server so it does not work with `ExternalIP` service so we need to make it different.